lib/app/endpoint/landing.rb

```diff
@@ -18,6 +18,21 @@ class Landing < Endpoint
           # Custom landing page intended to be overwritten for branded deployments
           erb :landing
         end
+
+        # Serve our public key in a JWKS
+        get '/jwks' do
+          if File.exist?('keyfile')
+            key_str = File.read('keyfile')
+            key = OpenSSL::PKey::RSA.new(key_str)
+          else
+            key = OpenSSL::PKey::RSA.generate(2048)
+            File.write('keyfile', key.export)
+          end
+
+          jwk = JSON::JWK.new(key.public_key)
+          jwks = JSON::JWK::Set.new(jwk)
+          jwks.to_json
+        end
```

Comment:
I don't love that this is in `landing.rb`, but I could not get the endpoint to work anywhere else. For example, copying and pasting the same method into `home.rb` or `oauth2_endpoints.rb` just results in 404s. If putting this endpoint in this file bothers people, I can continue to try to figure it out.